### PR TITLE
Natspec comments events

### PR DESCRIPTION
**Pull request by alpinechicken**
# Task:

Add missing naspec and comments

## Description

Added missing natspec and some comments

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x ] Document update

## How Has This Been Tested

Please describe how to test to verify the changes. Provide instructions so we can reproduce.

## FE Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] Added video recordings if it is a UI change

## User Facing Checklist

- [ ] I fully understand the user problem this PR is solving
- [ ] I know who the target user is for this PR and have a deep understanding of that user
- [ ] I have tried this flow thinking from the pov of the target user for this PR
- [ ] I (or working w someone on team) have scheduled a user test for this PR (if it is a large change)

### CODE REVIEW

File: packages/bull-vault/src/AuctionBull.sol [hunk 1]
```diff
@@ -143,6 +143,9 @@ contract AuctionBull is UniFlash, Ownable, EIP712 {
         uint256 oldDeltaLower, uint256 oldDeltaUpper, uint256 newDeltaLower, uint256 newDeltaUpper
     );
     event LeverageRebalance(bool isSellingUsdc, uint256 usdcAmount, uint256 wethLimitAmount);
+
+    event FullRebalance(uint256 crabAmount, uint256 clearingPrice);
```
Comment (aleone) [inline packages/bull-vault/src/AuctionBull.sol:147]: we prb need the other details (usdc amount, eth amt)

File: packages/bull-vault/src/AuctionBull.sol [hunk 2]
```diff
@@ -377,7 +382,7 @@ contract AuctionBull is UniFlash, Ownable, EIP712 {
                         require(currentPrice >= prevPrice, "AB7");
                     }
                     prevPrice = currentPrice;
-
+                    // transfer from order to contract
```
Comment (aleone) [inline packages/bull-vault/src/AuctionBull.sol:385]: we should fire events for traded orders probably

Comment (aleone) [reply packages/bull-vault/src/AuctionBull.sol:385]: (both buy and sell)

File: packages/bull-vault/src/BullStrategy.sol [hunk 3]
```diff
@@ -124,9 +122,8 @@ contract BullStrategy is ERC20, LeverageBull {
     }
 
     /**
-     * @notice deposit function that handle minting shares and depositing into the leverage component
-     * @dev this function assume the _from depositor already have _crabAmount
-     * @param _crabAmount amount of crab token
+     * @notice handle minting shares and rebalancing WETH/USDC leverage
```
Comment (aleone) [inline packages/bull-vault/src/BullStrategy.sol:125]: think this is the wrong description, right? doesn't rebalance

File: packages/bull-vault/src/BullStrategy.sol [hunk 4]
```diff
@@ -145,17 +142,20 @@ contract BullStrategy is ERC20, LeverageBull {
...
         require(_totalWethInEuler <= strategyCap, "BS2");
 
         IERC20(usdc).transfer(msg.sender, usdcBorrowed);
+
+        emit Deposit(msg.sender, _crabAmount);
     }
 
     /**
-     * @notice withdraw ETH from crab and euler by providing wPowerPerp, bull token and USDC to repay debt
+     * @notice withdraw ETH from crab and Euler by providing wPowerPerp, bull token, and USDC to repay debt
```
Comment (aleone) [inline packages/bull-vault/src/BullStrategy.sol:158]: maybe should note the fact that withdrawals are in ETH but deposits are in crab, in fairness maybe not the best decision :)

File: packages/bull-vault/src/BullStrategy.sol [hunk 5]
```diff
@@ -195,6 +200,10 @@ contract BullStrategy is ERC20, LeverageBull {
         emit RedeemCrabAndWithdrawEth();
```
Comment (aleone) [inline packages/bull-vault/src/BullStrategy.sol:200]: do we need anything in event, prb should add the crab and wpowerperp and eth redeemed?

File: packages/bull-vault/src/BullStrategy.sol [hunk 6]
```diff
@@ -180,6 +180,11 @@ contract BullStrategy is ERC20, LeverageBull {
         emit Withdraw(msg.sender, _bullAmount, wPowerPerpToRedeem);
```
Comment (aleone) [inline packages/bull-vault/src/BullStrategy.sol:180]: we probably need the eth received by caller so user can figure out pnl

File: packages/bull-vault/src/BullStrategy.sol [hunk 7]
```diff
@@ -206,6 +215,11 @@ contract BullStrategy is ERC20, LeverageBull {
         IERC20(wPowerPerp).transfer(msg.sender, IERC20(wPowerPerp).balanceOf(address(this)));
```
Comment (aleone) [inline packages/bull-vault/src/BullStrategy.sol:215]: should prb have event here? with eth deposited, crab received and wpowerperp?

File: packages/bull-vault/src/BullStrategy.sol [hunk 8]
```diff
@@ -206,6 +215,11 @@ contract BullStrategy is ERC20, LeverageBull {
...
     }
 
+    /**
+     * @notice close out Euler leverage position if contracts have been shut down
+     * @param wethToUniswap weth to repay to uniswap via auction contract
+     * @param shareToUnwind share of crab to redeem scaled by 1e18
+     */
```
Comment (aleone) [inline packages/bull-vault/src/BullStrategy.sol:222]: maybe event?

File: packages/bull-vault/src/FlashBull.sol [hunk 9]
```diff
@@ -181,6 +180,10 @@ contract FlashBull is UniFlash {
         IERC20(bullStrategy).transfer(msg.sender, IERC20(bullStrategy).balanceOf(address(this)));
```
Comment (aleone) [inline packages/bull-vault/src/FlashBull.sol:180]: no flashdeposit event, we need it

File: packages/bull-vault/src/FlashBull.sol [hunk 10]
```diff
@@ -181,6 +180,10 @@ contract FlashBull is UniFlash {
...
     }
 
+    /**
+     * @notice flash withdraw from strategy, receiving ETH, providing wSqueeth and USDC, and providing strategy tokens
+     * @param _params FlashWithdrawParams struct
+     */
```
Comment (aleone) [inline packages/bull-vault/src/FlashBull.sol:186]: think for flashwithdraw event we need to know the eth amount received by sender in end not just the params

File: packages/bull-vault/src/FlashBull.sol [hunk 11]
```diff
@@ -281,26 +286,26 @@ contract FlashBull is UniFlash {
...
-            uint256 wSqueethToMint = _depositedEthAmount.wmul(_strategyDebtAmount).wdiv(
-                _strategyCollateralAmount.add(_strategyDebtAmount.wmul(feeAdjustment))
-            );
-            uint256 fee = wSqueethToMint.wmul(feeAdjustment);
-            return (wSqueethToMint, fee);
-        }
-        uint256 wSqueethToMint =
-            _depositedEthAmount.wmul(_strategyDebtAmount).wdiv(_strategyCollateralAmount);
-        return (wSqueethToMint, 0);
+        uint256 feeRate = IController(IBullStrategy(bullStrategy).powerTokenController()).feeRate();
```
Comment (aleone) [inline packages/bull-vault/src/FlashBull.sol:301]: these changes were reverted, should be re-reverted

File: packages/bull-vault/src/FlashBull.sol [hunk 12]
```diff
@@ -323,11 +328,4 @@ contract FlashBull is UniFlash {
 
         return _amount;
     }
```
Comment (aleone) [inline packages/bull-vault/src/FlashBull.sol:330]: _getCrabVaultDetails() was reverted, need it back in

**Review by aleone — APPROVED** [review]
we gotta revert the few changes on the merge conflicts, but generally lgtm. would be great if could add a few more events

think LeverageBull prb needs events:
- _leverageDeposit
-depositAndBorrowFromLeverage
-auctionRepayAndWithdrawFromLeverage

events to ask KM on:

 - LeverageRebalance should maybe have the eth amt traded if possible
- executeCrabDeposit - could fire event here to give more info on the fullRebalance()
-executeLeverageComponentRebalancing - could fire event here to give more info on full rebalance

File: packages/bull-vault/src/AuctionBull.sol (hunk 1 above)
Comment (alpinechicken) [reply packages/bull-vault/src/AuctionBull.sol:147]: don't have usdc directly but adde d wPowerPerpAmount and isDepositingCrab

File: packages/bull-vault/src/AuctionBull.sol (hunk 2 above)
Comment (alpinechicken) [reply packages/bull-vault/src/AuctionBull.sol:385]: added

File: packages/bull-vault/src/BullStrategy.sol (hunk 3 above)
Comment (alpinechicken) [reply packages/bull-vault/src/BullStrategy.sol:127]: yeah definitely not right!

File: packages/bull-vault/src/BullStrategy.sol (hunk 4 above)
Comment (alpinechicken) [reply packages/bull-vault/src/BullStrategy.sol:163]: added descriptions

File: packages/bull-vault/src/BullStrategy.sol (hunk 5 above)
Comment (alpinechicken) [reply packages/bull-vault/src/BullStrategy.sol:213]: added

File: packages/bull-vault/src/BullStrategy.sol (hunk 8 above)
Comment (alpinechicken) [reply packages/bull-vault/src/BullStrategy.sol:239]: yep

File: packages/bull-vault/src/FlashBull.sol (hunk 9 above)
Comment (alpinechicken) [reply packages/bull-vault/src/FlashBull.sol:181]: added

File: packages/bull-vault/src/FlashBull.sol (hunk 10 above)
Comment (alpinechicken) [reply packages/bull-vault/src/FlashBull.sol:187]: yep added

**Pull request merged**